playwright/e2eTests/api-delegering.spec.ts

```diff
@@ -1,100 +1,228 @@
+
+  await apiDelegations.deleteDelegatedAPIs();
+
+  await apiDelegations.goToAccessToApiPageFromFrontPage();
+
+  await apiDelegations.delegateNewApiButton.click();
+  await apiDelegations.filterByAgencyButton.click();
+
+  await page.getByLabel(standardApiDetails.department).click();
+  await apiDelegations.applyButton.click();
+
+  const apiNames = [
+    'AuthorizedParties: Mainunit to Organization Maskinporten Schema for utviklingstester',
+    'Automation Regression',
+    'AuthorizedParties: Subunit to Organization Maskinporten Schema for utviklingstester',
+  ];
+
+  await apiDelegations.selectApiAccess(apiNames);
+
+  //  Select users that gets granted access
+  const organizations: [string, string][] = [
+    ['310661414', 'INTERESSANT KOMPATIBEL TIGER AS'],
+    ['310110914', 'LYKKELIG DRIFTIG APE'],
+    ['313259412', 'UROMANTISK ALTERNATIV KATT GÅS'],
+  ];
+
+  await apiDelegations.grantUserAccess(organizations);
+  await apiDelegations.confirmAccessGranted(apiNames, organizations);
+  await apiDelegations.verifyDelegatedRightsGiven(apiNames, organizations);
+});
+
+test('Verify that Tilgangsstyrer does NOT have access to API delegering panel', async ({
+  page,
+  login,
+  apiDelegations,
+}) => {
+  const testUser = '14824497789';
+  const reporteeWithoutAccess = 'BLÅVEIS SKRAVLETE';
+  await login.loginWithUser(testUser);
+  await login.chooseReportee(reporteeWithoutAccess);
+
+  await page.goto((process.env.BASE_URL as string) + '/ui/profile');
+
+  //Make sure the user is on the correct page
+  await expect(page.getByRole('heading', { name: `Profil for SKRAVLETE BLÅVEIS` })).toBeVisible();
+
+  //Verify that the user does not have access to the API delegering panel
+  await expect(apiDelegations.getApiAccessButton).not.toBeVisible();
+});
+
+// create new test that does exactly the same as the one above, but with a different user
+test('Verify reportee with Programmeringsgrensesnitt (API) role does have access to API-delegering panel', async ({
+  page,
+  login,
+  apiDelegations,
+}) => {
+  //Login as Tilgangsstyrer
+  const user = '14824497789';
+  const reporteeWithAPIRole = 'AKTVERDIG RETORISK APE';
+  await login.loginWithUser(user);
+  await login.chooseReportee(reporteeWithAPIRole);
+
+  await page.goto((process.env.BASE_URL as string) + '/ui/profile');
+
+  //Make sure the user is on the correct page
+  await expect(
+    page.getByRole('heading', { name: `Profil for ${reporteeWithAPIRole}` }),
+  ).toBeVisible();
+
+  //Verify that the user does have access to the API delegering panel
+  await expect(apiDelegations.getApiAccessButton).toBeVisible();
+});
+
+//Create a test that verifies that user gets an error when trying to delegate an API without the correct rights
+test('Verify reportee without API role does not have access to delegate an API', async ({
+  page,
+  login,
+}) => {
+  const user = '14824497789';
+  const reporteeWithoutAccess = 'AKTVERDIG RETORISK APE';
```

Comment:
We can use following code instead of creating new const for same user  
await login.loginWithUser(standardDaglUser.id);
await login.chooseReportee(standardDaglUser.reportee);